R/convenience-functions.R

```diff
@@ -140,6 +146,7 @@ transform_forecasts <- function(data,
     transformed_data[, observed := fun(observed, ...)]
     transformed_data[, scale := label]
     out <- rbind(original_data, transformed_data)
+    out <- suppressWarnings(suppressMessages(as_forecast(out)))
```

Comment:
if `as_forecast` had a copy arg this could be false I think? Only a small efficiency saving

Reply:
It doesn't :) I think copying by default is good and I wouldn't introduce another argument for this small efficiency gain. 

If we wanted to avoid copying here we could 
- change this line to `new_forecast()`
- think about whether `validate_forecast()` should have an option to copy/not to copy?